### PR TITLE
build(aio): update tsc, angular, cli and other dependencies

### PR DESCRIPTION
**Pull request by IgorMinar**

### CODE REVIEW

File: aio/src/app/custom-elements/code/code.component.spec.ts [hunk 1]
```diff
@@ -67,38 +67,40 @@ describe('CodeComponent', () => {
       expect(spans.length).toBeGreaterThan(0, 'formatted spans');
     });
 
-    function hasLineNumbers() {
+    async function hasLineNumbers(): Promise<boolean> {
       // presence of `<li>`s are a tell-tale for line numbers
+      await new Promise(resolve => hostComponent.codeComponent.codeFormatted.subscribe(resolve));
```
Comment (alfaproject) [inline aio/src/app/custom-elements/code/code.component.spec.ts]: Would `codeFormatted.toPromise()` work?

Comment (gkalpak) [reply aio/src/app/custom-elements/code/code.component.spec.ts]: I don't think so. AFAICT, `toPromise()` only resolves once the observable completes, but we just want to catch the next "emission". `codeFormatted.first().toPromise()` probably would though :grin:

**Comment by gkalpak** [comment]
- Travis build for upgrading to `rxjs[-compat]@6.0.0-beta.1`: [356384657][1]
- Travis build for switching to pipeable operators: [356385442][2]
- Travis build for removing `rxjs-compat`: [356397153][3]

Removing `rxjs-compat` fails due to `@angular/service-worker` and `@angular/material`.
`@angular/service-worker` issues should be fixed after upgrading to the latest version (#19795).
Not sure about `@angular/material`. 

[1]: https://travis-ci.org/angular/angular/builds/356384657
[2]: https://travis-ci.org/angular/angular/builds/356385442
[3]: https://travis-ci.org/angular/angular/builds/356397153

**Comment by IgorMinar** [comment]
Material release that won't depend on rxjs-compat is expect today/tomorrow.

**Comment by gkalpak** [comment]
As discussed, removed the last commit (94d443d63), keeping `rxjs-compat` until we can upgrade to versions of `@angular/material` and `@angular/service-worker` that don't need it.

**Comment by gkalpak** [comment]
For reference:
Some examples e2e tests are failing when run against the locally built packages, because `angular/in-memory-web-api` is not compatible with RxJS@6. Here is the PR for upgrading `angular/in-memory-web-api`: https://github.com/angular/in-memory-web-api/pull/185/files

**Comment by IgorMinar** [comment]
I worked with @wardbell to update angular-in-memory-api and we now have a 0.6 release out. I update aio examples to that and pushed the change to this pr.

**Comment by IgorMinar** [comment]
This PR is now green! thanks @gkalpak 

once the material release with native support rxjs v6 is out we can upgrade to that. in the meantime let's finish the sw upgrade. Once both of these are upgraded we can remove the rxjs-compat package!

**Comment by IgorMinar** [comment]
george - can you please approve this PR?

File: aio/tools/ng-packages-installer/index.js [hunk 2]
```diff
@@ -169,7 +169,6 @@ class NgPackagesInstaller {
     // FIXME: Temporarily use RxJS from root `node_modules/`.
     if (peerDependencies.rxjs) {
       peerDependencies.rxjs = `file:${ANGULAR_ROOT_DIR}/node_modules/rxjs`;
```
Comment (gkalpak) [inline aio/tools/ng-packages-installer/index.js]: Are there still local changes in `angular/node_modules/rxjs/`, or can we get rid of this too?

**Review by gkalpak — APPROVED** [review]

**Pull request closed without merging**